Composer/plugins/azurePublish/src/index.ts

```diff
@@ -0,0 +1,277 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import path from 'path';
+
+import { BotProjectDeploy } from '@bfc/libs/bot-deploy';
+import { v4 as uuid, v5 as hash } from 'uuid';
+import { copy, emptyDir, readJson, pathExists, writeJson, stat } from 'fs-extra';
+
+interface CreateAndDeployResources {
+  name: string;
+  location: string;
+  environment: string;
+  subscriptionID: string;
+  luisAuthoringKey?: string;
+  luisAuthoringRegion?: string;
+  appPassword: string;
+  accessToken: string;
+  graphToken: string;
+}
+
+interface PublishConfig {
+  customizeConfiguration: CreateAndDeployResources;
```

Comment:
I don't love this. What is the reason to create this subobject?

Reply:
I will remove this and rename the name field into profileName.

Reply:
done